diff --git a/controller/core/is_auth.php b/controller/core/is_auth.php
index d2a0b2a..531dfec 100644
--- a/controller/core/is_auth.php
+++ b/controller/core/is_auth.php
@@ -6,7 +6,7 @@ session_start();
 // If test fails, send the user to login.php and prevent rest of page being shown.
 if (!isset($_SESSION["is_auth"]) || !isset($_SESSION["access_level"])) {
 	session_destroy();
-	printf( '<p class="red">Access NOT authenticated</p>' ); header("refresh:2; url=../view/front.php");
+	printf( '<p class="red" style="color:red">Access NOT authenticated</p>' ); header("refresh:2; url=../view/front.php");
 	exit;
 }
 else if (isset($_REQUEST['logout']) && $_REQUEST['logout'] == "true") {
diff --git a/controller/front_post.php b/controller/front_post.php
index e7fc426..4ccc810 100644
--- a/controller/front_post.php
+++ b/controller/front_post.php
@@ -4,26 +4,26 @@
 
 	$t_user_table = 'mantis_user_table';
 	require_once('core/date_time.php');
-	$_SESSION['time']     = getDateTime();
-	$_SESSION['username'] = $_POST['username'];
-	$_SESSION['password'] = $_POST['password'];
-	$_SESSION['assembly'] = $_POST['assembly'];
-	$_SESSION['revision'] = $_POST['revision'];
-	$_SESSION['format'] = $_POST['format'];
-	$_SESSION['format_example'] = $_POST['format_example'];
-	$_SESSION['sale_order'] = $_POST['sale_order'];
-	$_SESSION['ss_key'] = 0;
+	$_SESSION['time']     			= getDateTime();
+	$_SESSION['username'] 			= $mysqli->real_escape_string($_POST['username']);
+	$_SESSION['password'] 			= $mysqli->real_escape_string($_POST['password']);
+	$_SESSION['assembly_number'] 	= $mysqli->real_escape_string($_POST['assembly_number']);
+	$_SESSION['revision'] 			= $mysqli->real_escape_string($_POST['revision']);
+	$_SESSION['format'] 			= $mysqli->real_escape_string($_POST['format']);
+	$_SESSION['format_example'] 	= $mysqli->real_escape_string($_POST['format_example']);
+	$_SESSION['sale_order'] 		= $mysqli->real_escape_string($_POST['sale_order']);
+	$_SESSION['customer'] 			= $mysqli->real_escape_string($_POST['customer']);
+	$_SESSION['ss_key'] 			= 0;
 
 	echo 'username : ' . $_POST['username']. '<br/>';
 	echo 'password : ' . $_POST['password']. '<br/>';
 	echo 'format : ' . $_POST['format']. '<br/>';
 	echo 'format_example : ' . $_POST['format_example']. '<br/>';
-	echo 'sale_order : ' . $_POST['sale_order']. '<br/>';
-	$t_username 		= $mysqli->real_escape_string($_SESSION['username']);
-	$t_password 		= $mysqli->real_escape_string($_SESSION['password']);
-		$t_password_copy 		= $t_password;
-	$t_format 	= $mysqli->real_escape_string($_SESSION['format']);
-	$t_format_example 		= $mysqli->real_escape_string($_SESSION['format_example']);
-	$t_sale_order 		= $mysqli->real_escape_string($_SESSION['sale_order']);
+	
+	$t_username 			= $_SESSION['username'];
+	$t_password 			= $_SESSION['password'];
+		$t_password_copy 	= $t_password;
+	$t_format 				= $_SESSION['format'];
+	$t_format_example 		= $_SESSION['format_example'];
 	require_once('core/access_control.php');
 ?>
diff --git a/js/front.js b/js/front.js
index f78c03e..62316b3 100644
--- a/js/front.js
+++ b/js/front.js
@@ -152,6 +152,7 @@ $('#customer .typeahead').bind('typeahead:select', function(ev, suggestion) {
               }
           }
       });
+      $('#assembly .typeahead').focus();
   })
   .fail(function(jqXHR, textStatus, errorThrown){
     console.log(jqXHR, textStatus, errorThrown);
@@ -160,6 +161,8 @@ $('#customer .typeahead').bind('typeahead:select', function(ev, suggestion) {
 
 $('#assembly .typeahead').bind('typeahead:select', function(ev, suggestion) {
   console.log('Selection: ' + JSON.stringify(suggestion));
+  var myData = JSON.stringify({"revision": suggestion.eg});
+  $("#result").append(myData + "<br/>");
 });
 /*var $t_format_example = $( this ).find( 'input:hidden' );
   var foo = $('input.typeahead.tt-input').val();
diff --git a/view/index.php b/view/index.php
index 1ff6b1f..b7af983 100644
--- a/view/index.php
+++ b/view/index.php
@@ -1,4 +1,3 @@
-<?php $t_str = "myString";?>
 <head>
     <link href="../bootstrap/css/bootstrap.css" rel="stylesheet">
     <script type="text/javascript" src="../plugin/jquery/jquery-1.11.3.min.js"></script>
diff --git a/view/serialscan_main_view.php b/view/serialscan_main_view.php
index e9aceb9..3335768 100644
--- a/view/serialscan_main_view.php
+++ b/view/serialscan_main_view.php
@@ -17,15 +17,16 @@
 	require_once('../controller/core/is_auth.php');
 
 	$t_arr = [];
-	$t_arr['assembly_number'] = $_SESSION['assembly_number'];
+	$t_arr['assembly'] = $_SESSION['assembly'];
 	$t_arr['revision'] = $_SESSION['revision'];
+	$t_arr['customer'] = $_SESSION['customer'];
 	$t_arr['sale_order'] = $_SESSION['sale_order'];
 	$t_arr['format'] = $_SESSION['format'];
 	$t_arr['format_example'] = $_SESSION['format_example'];
 	$t_arr['access_level'] = $_SESSION['access_level'];
 	$t_arr['username'] = $_SESSION['username'];
 	$t_arr['time'] = $_SESSION['time'];
-	$t_arr['primary_key'] = $_SESSION['primary_key'];
+	$t_arr['ss_key'] = $_SESSION['ss_key'];
 
 	$jsonString = json_encode($t_arr, JSON_PRETTY_PRINT);
 
