diff --git a/controller/scan_proc.php b/controller/scan_proc.php
index 21c30b2..d97ae8c 100644
--- a/controller/scan_proc.php
+++ b/controller/scan_proc.php
@@ -1,11 +1,13 @@
 <?php
     require_once('../model/dbi_con.php');
+    require_once('../controller/core/is_auth.php');
     
     if(isset($_POST['qr']))
 {
-    $qr = $_POST['qr'];
+    $_SESSION['ss_key'] += 1;
+    $qr                     = $_POST['qr'];
     if (is_scalar($qr))
-        echo $qr;
+       echo $_SESSION['ss_key'] . ". " . $qr;
         // echo json_encode($qr, JSON_FORCE_OBJECT);
         // echo json_encode($qr, JSON_PRETTY_PRINT);
     
diff --git a/js/front.js b/js/front.js
index c2597fa..fe3338d 100644
--- a/js/front.js
+++ b/js/front.js
@@ -35,6 +35,7 @@
 
 (function(){
 var jqDeferred = $.ajax({
+  type: "POST",
   dataType: "json",
   url: "../model/json_db/customer.php"});
   jqDeferred.then( function(data) {
diff --git a/model/json_db/assembly.php b/model/json_db/assembly.php
index b7c7f12..cbc9ac7 100644
--- a/model/json_db/assembly.php
+++ b/model/json_db/assembly.php
@@ -12,7 +12,7 @@ require_once('../dbi_con.php');
 		
 		if ($num_rows){
 			$t_format_arr = [];
-			while ( ($row=$result->fetch_assoc()) !== null ) {
+			while ($row=$result->fetch_assoc()) {
 				$t_format_arr[] = $row;
 			}
 			$result->free();
diff --git a/model/json_db/customer.php b/model/json_db/customer.php
index b53b0f4..a2761ab 100644
--- a/model/json_db/customer.php
+++ b/model/json_db/customer.php
@@ -17,7 +17,7 @@ require_once('../dbi_con.php');
 		
 	if ($num_rows){
 		$t_format_arr = [];
-		while ( ($row=$result->fetch_assoc()) !== null ) {
+		while ($row=$result->fetch_assoc()) {
 			$t_format_arr[] = $row;
 		}
 		$result->free();
diff --git a/model/json_db/format.php b/model/json_db/format.php
index e1838a3..f68325d 100644
--- a/model/json_db/format.php
+++ b/model/json_db/format.php
@@ -13,7 +13,7 @@ if($_POST["assembly_id"]){
 
 	if ($num_rows){
 		$t_format_arr = [];
-		while ( ($row=$result->fetch_assoc()) !== null ) {
+		while ($row=$result->fetch_assoc()) {
 			$t_format_arr[] = $row;
 		}
 		$result->free();
diff --git a/model/json_db/revision.php b/model/json_db/revision.php
index 118db2a..04d9f75 100644
--- a/model/json_db/revision.php
+++ b/model/json_db/revision.php
@@ -10,7 +10,7 @@ require_once('../dbi_con.php');
 
 		if ($num_rows){
 			$t_format_arr = [];
-			while ( ($row=$result->fetch_assoc()) !== null ) {
+			while ($row=$result->fetch_assoc()) {
 				$t_format_arr[] = $row;
 			}
 			$result->free();
diff --git a/view/serialscan_main_view.php b/view/serialscan_main_view.php
index f16845d..becb192 100644
--- a/view/serialscan_main_view.php
+++ b/view/serialscan_main_view.php
@@ -26,7 +26,8 @@
 	$t_arr['access_level'] 		= $_SESSION['access_level'];
 	$t_arr['username'] 			= $_SESSION['username'];
 	$t_arr['time'] 				= $_SESSION['time'];
-	$t_arr['ss_key'] 			= $_SESSION['ss_key'];
+	$t_arr['ssId']				= session_id();
+	$t_arr['sessionId']			= str_repeat('*', max(1, strlen($t_arr['ssId']) - 4)) . substr($t_arr['ssId'], strlen($t_arr['ssId']) - 4, strlen($t_arr['ssId']));
 
 	$jsonString = json_encode($t_arr, JSON_PRETTY_PRINT);
 
@@ -147,8 +148,6 @@
 	<script type="text/javascript">
 	var jsonData =<?php echo $jsonString?>;
 	(function(){
-		console.log(jsonData);
-		console.log(jsonData.username);
 	    // Grab the template script $("homeview-template").html()
 		var myInfo = document.getElementById("homeview-template").innerHTML;
 
@@ -187,7 +186,7 @@
 			    access_level: jsonData.access_level,
 			    username: jsonData.username,
 			    time: jsonData.time,
-			    sessionId: 'md5#0001'
+			    sessionId: "sessionId: " + jsonData.sessionId
 			}
 		};
 		Handlebars.registerHelper("required", function(){
